File: projects/lostengine/lost/lua/lost/io/rg/DepthTest.lua
module("lost.io", package.seeall)

require("lost.io.Loader")

function Loader:DepthTest(definition)
  local result = lost.rg.DepthTest.create(definition[1])
  
  if definition["active"] then
    result.active = definition["active"]
  end
  
  return result
end

File: projects/lostengine/lost/rg/DepthTest.cpp
#include "lost/rg/DepthTest.h"
#include "lost/gl/Context.h"

namespace lost
{
namespace rg
{

DepthTest::DepthTest(bool inEnable)
:enable(inEnable)
{
  name = "depthTest";
}

DepthTestPtr DepthTest::create(bool inEnable)
{
  return DepthTestPtr(new DepthTest(inEnable));
}

void DepthTest::process(gl::ContextPtr ctx)
{
  if(!active) return;
  ctx->depthTest(enable);
}

}
}